src/sync/sys/windows/net.rs

```diff
@@ -153,6 +176,7 @@ impl PipeListener {
 
     pub fn close(&self) -> Result<()> {
         // release the ConnectNamedPipe thread by signaling the event and clean up event handle
+        self.shutting_down.store(true, Ordering::SeqCst);
```

Comment:
we need both of these.  `set_event` is what signals the thread that is suspended by `GetOverlappedResult`.  The `close_handle` ensures that we don't leak the handle now that we've shut down.  I might be able to move that to a `Drop` function for the `pipelistener` since I guess we could call accept again later after closing? I would need to think about that a bit more.